### ISSUE TITLE
Report Reviewer #1

### ISSUE THREAD

**Issue by MarieEtienne** [computorg/published-202402-lefort-peerannot#4]
Associate Editor: *MP Etienne*
Reviewer : [name](url) (chose to lift his/her anonymity / remain anonymous)

## Reviewer: Reviewing history

- Paper submitted *September, 8 2023*
- Reviewer invited *September, 20 2023*
- Review 1 received *October, 24, 2023*
- Paper revised *date*
- Reviewer invited *date*
- Review 2 received *date*
- Paper conditionally accepted *date*

### Summary Of Contributions:

This very clear, well-constructed and documented article details methods for pre-processing crowd-sourced annotation for training classification models.

The authors publish an easy-to-use open source library implementing the algorithms described. They include a few datasets with examples.

They also propose a convention for structuring model training datasets, thus improving the compatibility of tools with different models.

This work is in the spirit of open research. By publishing a practical library and seeking to homogenize practices, the authors are helping to facilitate the work of other players in the community. It's a useful piece of work, and a worthy modern approach.
Strengths And Weaknesses:

The library is particularly well-developed and packaged. Documentation is clear (both online and in the commands themselves). It works perfectly.

The paper is also very clear and well constructed. It does, however, suffer from a few formatting problems and some bloat in the part including code.
### Changes And Questions:

All remarks below are required. The first point could be addressed by arguing the need to provide the graphics code in the article flow.
Python Code not very useful in the article

In general, I find most python code in the paper not bringing much value : It does not demonstrate usage of API of the library, but rather some standard matplotlib code to produce the graphs. I feel that the graphs speak from themselves and the readers don't really need to know the code that produced it at first. For the few Python code using internal function of the library, you may put this in some supplementary materials, or as sample code in the library and refer to it in the paper.

I feel it may ease the readability of the paper.

This remark does not apply to CLI / Bash commands, which I find useful.
Code formatting and long lines

Several long lines are cut in the PDF output, both for Python and CLI commands. Examples :

    2.2 - Page 6 : "plane", 8 ...
    2.2 - Page 7 - end : "opencountry", ...
    3. - Page 11 - end: --feedback=...
    ... please double-check for other code (CLI & Python)

### Missing graphs

    5.1.1 & 5.1.2 - Page 32 : The following text appears 3 times instead of a proper graph : Unable to display output for mime type(s): text/html

### Misc remarks

    Figure 6 - Page 13 : Please label the vertical axis "ground truth", or similar
    Figure 13 : Please explain more in detail what a pair plot is / how to read it. The top-left plot seem to lack a vertical axis / labeling
    6 Conclusion - Page 37 : The conclusion is attached to the bibliography, without proper space or title to separate them.

###Comments On Reproducibility:
 Reproducibility: Yes

The library is well packaged and works perfectly out of the box. The CLI commands are very clear and online documentation (--help) is very good.

### Some minor suggestions :

    Enhance the visibility of the documentation: Googling peerannot returns the personal page of the first author and the [pip page](https://pypi.org/project/peerannot/). The latter lacks a link to the homepage (either Github project or github.io page) and extended description.

    Enforce python version compatibility : I was able to run the library using python3.10, but was not sure which version is supported. You may enforce the min/max versions of python in the documentation and [setup config](https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#python-requires)

    Cifar: It was not clear in the documentation (but more clear in the paper) that the cifar10H folder was provided in the folder datasets and required to clone the repository. I first understood I had to download & install it myself locally from official Cifar website. Could you make it more obvious in the documentation ?

    job_xx.py : The folder datasets contains several job_x.py that seem to be private files, not used, referring to private absolute home folder the author.

    Heavy dependencies : The dependencies are quite heavy, especially vision. It may be useful to make torch dependencies optional, for those only waiting to use aggregate & identify only and having already their own training pipeline. Don't bother to do it if the code is too much dependent on this though.

>  **authors' answer:**
